### PR TITLE
Add task runner documentation

### PR DESCRIPTION
**Pull request by tomi**

### CODE REVIEW

File: docs/hosting/securing/hardening-task-runners.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+contentType: howto
+---
+
+# Hardening task runners
+
+[Task runners](/hosting/configuration/task-runners/) are responsible for executing code from the [Code node](/integrations/builtin/core-nodes/n8n-nodes-base.code/). While Code node executions are secure, you can follow these recommendations to further harden your task runners.
+
+## Run task runners as sidecars in external mode
+
+To increase the isolation between the core n8n process and code in the Code node, run task runners in [external mode](/hosting/configuration/task-runners#setting-up-external-mode). External task runners launch as separate containers, providing a sandboxed environment to execute the JavaScript defined in the Code node.
```
Comment (tomi) [inline docs/hosting/securing/hardening-task-runners.md:14]: Both modes provide a sandboxed execution environment. Could we rephrase this to something like

```suggestion
To increase the isolation between the core n8n process and code in the Code node, run task runners in [external mode](/hosting/configuration/task-runners#setting-up-external-mode). External task runners launch as separate containers, providing a fully isolated environment to execute the JavaScript defined in the Code node.
```
?

Meaning `internal` mode is partially isolated (i.e. they share the same host/container and filesystem), and `external` is fully isolated (separate hosts/containers, no filesystem access)

File: docs/hosting/configuration/environment-variables/task-runners.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+| `N8N_RUNNERS_TASK_TIMEOUT` | Number | `60` | How long (in seconds) a task can take to complete before the task aborts and the runner restarts. Must be greater than 0. |
+| `N8N_RUNNERS_HEARTBEAT_INTERVAL` | Number | `30` | How often (in seconds) the runner must send a heartbeat to the broker, else the task aborts and the runner restarts. Must be greater than 0. |
+
+
+## Task runner launcher environment variables
+
+| Variable | Type  | Default  | Description |
+| :------- | :---- | :------- | :---------- |
+| `N8N_RUNNERS_LAUNCHER_LOG_LEVEL` | Enum string: `debug`, `info`, `warn`, `error` | `info` | Which log messages to show. |
+| `N8N_RUNNERS_AUTH_TOKEN` | String | Random string | Shared secret used to authenticate to n8n. |
```
Comment (tomi) [inline docs/hosting/configuration/environment-variables/task-runners.md:40]: I actually set the default wrong. In the n8n instance it has a default value, but the launcher doesn't have one

```suggestion
| `N8N_RUNNERS_AUTH_TOKEN` | String | - | Shared secret used to authenticate to n8n. |
```

File: docs/hosting/configuration/environment-variables/task-runners.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+| `N8N_RUNNERS_AUTO_SHUTDOWN_TIMEOUT` | Number | `15` | The number of seconds to wait before shutting down an idle runner. |
+| `N8N_RUNNERS_TASK_BROKER_URI` | String | `http://127.0.0.1:5679` | The URI of the task broker server (n8n instance). |
+| `N8N_RUNNERS_LAUNCHER_HEALTH_CHECK_PORT` | Number | `5680` | Port for the launcher's health check server. |
+| `N8N_RUNNERS_MAX_PAYLOAD` | Number | `1 073 741 824` | Maximum payload size in bytes for communication between a task broker and a task runner. |
+| `N8N_RUNNERS_MAX_OLD_SPACE_SIZE` | String |  | The `--max-old-space-size` option to use for a task runner (in MB). By default node.js will determine it based on the available memory. |
```
Comment (tomi) [inline docs/hosting/configuration/environment-variables/task-runners.md:45]: Not available for the launcher

```suggestion
```

File: docs/hosting/configuration/task-runners.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+| `command`       | `["/usr/local/bin/task-runner-launcher", "javascript"]` |
+| `livenessProbe` | `GET /healthz`, port `5680`                             |
+
+Set the following environment variables for the container, adjusted to fit your needs:
+
+| Environment variables | Description |
+| ------ | ----- |
+| `N8N_RUNNERS_AUTH_TOKEN=<random secure shared secret>` | The shared secret the task runner uses to connect to the broker. |
+| `N8N_RUNNERS_MAX_CONCURRENCY=5` | The number of concurrent tasks the runner can execute. |
+| `N8N_RUNNERS_SERVER_ENABLED=true` | Enable the health check server on the runner. |
```
Comment (tomi) [inline docs/hosting/configuration/task-runners.md:79]: Actually we don't need this
```suggestion
```

File: docs/hosting/securing/hardening-task-runners.md (hunk 1 above)
Comment (imchairmanm) [reply docs/hosting/securing/hardening-task-runners.md:14]: Yep, that works for me! Thanks for clarifying.

**Review by imchairmanm — APPROVED** [review]

File: docs/hosting/configuration/task-runners.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+| `N8N_RUNNERS_MODE=external`                            | Use task runners in external mode.                         |
+| `N8N_RUNNERS_AUTH_TOKEN=<random secure shared secret>` | A shared secret task runners use to connect to the broker. |
+
+For full list of environment variables see [task runner environment variables](/hosting/configuration/environment-variables/task-runners/).
+
+### Configuring task runners in external mode
+
+The task runner comes bundled within the n8n Docker image. The Docker image also includes the task runner launcher.
+
+The launcher can start the runner on-demand, which means lower memory usage when there's no work needed, but a short delay (few hundred ms) in cold-start. The launcher also monitors the launcher and restarts it in case of infinite loops or other issues.
```
Comment (ivov) [inline docs/hosting/configuration/task-runners.md:64]:
```suggestion
The launcher can start the runner on-demand, which means lower memory usage when there's no work needed, but a short delay (few hundred ms) in cold-start. The launcher also monitors the runner and restarts it in case of infinite loops or other issues.
```
